src/DcmTrajectoryGenerator.cpp

```diff
@@ -0,0 +1,652 @@
+/**
+ * @file DcmTrajectoryGenerator.cpp
+ * @author Giulio Romualdi
+ * @copyright 2018 iCub Facility - Istituto Italiano di Tecnologia
+ *            Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+ * @date 2018
+ */
+
+// std
+#include <math.h>
+#include <vector>
+#include <iostream>
+
+// eigen
+#include <Eigen/Dense>
+
+// iDynTree
+#include "iDynTree/Core/VectorFixSize.h"
+#include "iDynTree/Core/EigenHelpers.h"
+
+#include "DcmTrajectoryGenerator.h"
+
+GeneralSupportTrajectory::GeneralSupportTrajectory(const double &startTime, const double &endTime)
+{
+  // set the general support trajectory domain
+  m_trajectoryDomain = std::make_pair(startTime, endTime);
+}
+
+bool GeneralSupportTrajectory::timeBelongsToDomain(const double &t)
+{
+  // check if m_startTime <= t <=  m_endTime
+  double startTime = std::get<0>(m_trajectoryDomain);
+  double endTime = std::get<1>(m_trajectoryDomain);
+  
+  if ((t >= startTime) && (t <= endTime))
+    return true;
+
+  std::cerr << "[GENERAL SUPPORT TRAJECTORY] the time t: " << t << " does not belong to the trajectory domain" << std::endl; 
+  return false;
+}
+
+const std::pair<double, double> &GeneralSupportTrajectory::getTrajectoryDomain() const
+{
+  return m_trajectoryDomain;
+}
+
+SingleSupportTrajectory::SingleSupportTrajectory(const double &startTime,
+						 const double &endTime,
+						 const double &t0,
+						 const double &stepDuration,
```

Comment:
I don't understand why you need the duration here. Is't it just ``endTime - startTime``?

Reply:
Unfortunately the `stepDuration` variable is not  `endTime - startTime`.  
The single support (SS) trajectory is evaluated under the hypothesis of instantaneous transitions between left and right SS phases [no double support (DS)]. Then the DS phase is evaluated using an heuristic tecnique.  
So the `stepDuration` variable is the temporal distance between the `t0` value of two consecutive SS trajectory.
I attach a simple scheme to clarify the concept.
![dcmduration](https://user-images.githubusercontent.com/16744101/35397127-efbfa0a8-01ee-11e8-8eb1-897c9458c5ce.png)
The orange segment represents the DS phase while the green one the SS phase.
You can found further information [here](http://ieeexplore.ieee.org/document/6943128/).

Reply:
After a discussion with @S-Dafarra we decide to change the `SingleSupportTrajectory` class in order to avoid future misunderstanding with the "temporal" variables.